x/evm/keeper/grpc_query_test.go

```diff
@@ -0,0 +1,487 @@
+								Removed:     false,
+							},
+						},
+					},
+					{
+						Hash: ethcmn.BytesToHash([]byte("tx_hash_1")).String(),
+						Logs: []*types.Log{
+							{
+								Address:     suite.address.String(),
+								Topics:      []string{ethcmn.BytesToHash([]byte("topic")).String()},
+								Data:        []byte("data"),
+								BlockNumber: 1,
+								TxHash:      ethcmn.BytesToHash([]byte("tx_hash_1")).String(),
+								TxIndex:     1,
+								BlockHash:   ethcmn.BytesToHash([]byte("block_hash")).String(),
+								Index:       0,
+								Removed:     false,
+							},
+							{
+								Address:     suite.address.String(),
+								Topics:      []string{ethcmn.BytesToHash([]byte("topic_1")).String()},
+								Data:        []byte("data_1"),
+								BlockNumber: 1,
+								TxHash:      ethcmn.BytesToHash([]byte("tx_hash_1")).String(),
+								TxIndex:     1,
+								BlockHash:   ethcmn.BytesToHash([]byte("block_hash")).String(),
+								Index:       0,
+								Removed:     false,
+							},
+						},
+					},
+				}
+
+				suite.app.EvmKeeper.SetLogs(suite.ctx, ethcmn.BytesToHash([]byte("tx_hash_0")), types.LogsToEthereum(expLogs[0].Logs))
+				suite.app.EvmKeeper.SetLogs(suite.ctx, ethcmn.BytesToHash([]byte("tx_hash_1")), types.LogsToEthereum(expLogs[1].Logs))
+
+				req = &types.QueryBlockLogsRequest{
+					Hash: hash.String(),
+				}
+			},
+			true,
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
+			suite.SetupTest() // reset
+
+			tc.malleate()
+			ctx := sdk.WrapSDKContext(suite.ctx)
+			res, err := suite.queryClient.BlockLogs(ctx, req)
+
+			if tc.expPass {
+				suite.Require().NoError(err)
+				suite.Require().NotNil(res)
+
+				suite.Require().Equal(expLogs, res.TxLogs)
+			} else {
+				suite.Require().Error(err)
+			}
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestQueryBlockBloom() {
+	var (
+		req      *types.QueryBlockBloomRequest
+		expBloom []byte
+	)
+
+	testCases := []struct {
+		msg      string
+		malleate func()
+		expPass  bool
+	}{
+		{"bloom bytes not found for height",
+			func() {},
+			false,
+		},
+		{
+			"success",
+			func() {
+				bloom := ethtypes.BytesToBloom([]byte("bloom"))
+				expBloom = bloom.Bytes()
+				suite.ctx = suite.ctx.WithBlockHeight(1)
+				suite.app.EvmKeeper.SetBlockBloom(suite.ctx, 1, bloom)
+			},
+			true,
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
+			suite.SetupTest() // reset
+
+			tc.malleate()
+			ctx := sdk.WrapSDKContext(suite.ctx)
+			ctx = metadata.AppendToOutgoingContext(ctx, grpctypes.GRPCBlockHeightHeader, fmt.Sprintf("%d", suite.ctx.BlockHeight()))
+			res, err := suite.queryClient.BlockBloom(ctx, req)
+
+			if tc.expPass {
+				suite.Require().NoError(err)
+				suite.Require().NotNil(res)
+
+				suite.Require().Equal(expBloom, res.Bloom)
+			} else {
+				suite.Require().Error(err)
+			}
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestQueryParams() {
```

Comment:
should we have a test where there is a non-empty param request?

Reply:
there are no args on the request